wicket-core/src/main/java/org/apache/wicket/pageStore/DiskPageStore.java

```diff
@@ -0,0 +1,717 @@
+						"DiskPageStore not configured for serialization");
+				}
+				data = serializer.serialize(page);
+				type = page.getClass().getName();
+			}
+
+			diskData.savePage(page.getPageId(), type, data);
+		}
+	}
+
+	/**
+	 * 
+	 * @param context
+	 * @param create
+	 * @return the session entry
+	 */
+	protected DiskData getDiskData(final IPageContext context, final boolean create)
+	{
+		SessionAttribute attribute = getSessionAttribute(context, create);
+
+		if (!create && attribute == null)
+		{
+			return null;
+		}
+
+		return getDiskData(attribute.identifier, create);
+	}
+
+	protected DiskData getDiskData(String sessionIdentifier, boolean create)
+	{
+		if (!create)
+		{
+			return diskDatas.get(sessionIdentifier);
+		}
+
+		DiskData data = new DiskData(this, sessionIdentifier);
+		DiskData existing = diskDatas.putIfAbsent(sessionIdentifier, data);
+		return existing != null ? existing : data;
+	}
+
+	protected SessionAttribute getSessionAttribute(IPageContext context, boolean create)
+	{
+		SessionAttribute attribute = context.getSessionAttribute(KEY);
+		if (attribute == null && create)
+		{
+			attribute = new SessionAttribute(applicationName, context.getSessionId());
+			context.setSessionAttribute(KEY, attribute);
+		}
+		return attribute;
+	}
+
+	/**
+	 * Load the index
+	 */
+	@SuppressWarnings("unchecked")
+	private void loadIndex()
+	{
+		File storeFolder = folders.getBase();
+
+		File index = new File(storeFolder, INDEX_FILE_NAME);
+		if (index.exists() && index.length() > 0)
+		{
+			try
+			{
+				InputStream stream = new FileInputStream(index);
+				ObjectInputStream ois = new ObjectInputStream(stream);
+				try
+				{
+					diskDatas.clear();
+
+					for (DiskData diskData : (List<DiskData>)ois.readObject())
+					{
+						diskData.pageStore = this;
+						diskDatas.put(diskData.sessionIdentifier, diskData);
+					}
+				}
+				finally
+				{
+					stream.close();
+					ois.close();
+				}
+			}
+			catch (Exception e)
+			{
+				log.error("Couldn't load DiskPageStore index from file " + index + ".", e);
+			}
+		}
+		Files.remove(index);
+	}
+
+	/**
+	 * 
+	 */
+	private void saveIndex()
+	{
+		File storeFolder = folders.getBase();
+		if (storeFolder.exists())
+		{
+			File index = new File(storeFolder, INDEX_FILE_NAME);
+			Files.remove(index);
+			try
+			{
+				OutputStream stream = new FileOutputStream(index);
+				ObjectOutputStream oos = new ObjectOutputStream(stream);
+				try
+				{
+					List<DiskData> list = new ArrayList<>(diskDatas.size());
+					for (DiskData diskData : diskDatas.values())
+					{
+						if (diskData.sessionIdentifier != null)
+						{
+							list.add(diskData);
+						}
+					}
+					oos.writeObject(list);
+				}
+				finally
+				{
+					stream.close();
+					oos.close();
+				}
+			}
+			catch (Exception e)
+			{
+				log.error("Couldn't write DiskPageStore index to file " + index + ".", e);
+			}
+		}
+	}
+
+	@Override
+	public Set<String> getContextIdentifiers()
```

Comment:
Is this the session ids ? I think it is. Should we use the same language consistently ? It is a bit confusing once to see `context` and other times `session`